irohad/util/utility_client.cpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "util/utility_client.hpp"
+
+#include <map>
+
+#include "util/proto_status_tools.hpp"
+
+#include <grpc++/grpc++.h>
+#include <boost/optional.hpp>
+#include "common/bind.hpp"
+#include "logger/logger.hpp"
+#include "network/impl/grpc_channel_builder.hpp"
+#include "utility_endpoint.grpc.pb.h"
+
+using namespace iroha::utility_service;
+
+using iroha::operator|;
+
+static_assert(logger::kIsFmtInternalSupported<
+                  iroha::utility_service::proto::Status> == false,
+              "qwe");
+
+struct UtilityClient::StubHolder {
+  StubHolder(const std::string &address)
+      : channel_(grpc::CreateCustomChannel(
+            address,
+            ::grpc::InsecureChannelCredentials(),
+            iroha::network::details::getChannelArguments<
+                proto::UtilityService_v1>())),
+        stub_(proto::UtilityService_v1::NewStub(channel_)) {}
+
+  std::shared_ptr<grpc::Channel> channel_;
+  std::unique_ptr<proto::UtilityService_v1::StubInterface> stub_;
+};
+
+UtilityClient::UtilityClient(const std::string irohad_address,
+                             logger::LoggerPtr log)
+    : log_(std::move(log)),
+      stub_holder_(std::make_unique<StubHolder>(irohad_address)) {}
+
+UtilityClient::~UtilityClient() = default;
+
+bool UtilityClient::waitForServerReady(
+    std::chrono::milliseconds timeout) const {
+  auto state = stub_holder_->channel_->GetState(true);
+  auto state_is_ready = [&state] {
+    return state == grpc_connectivity_state::GRPC_CHANNEL_READY;
+  };
+  auto deadline = std::chrono::system_clock::now() + timeout;
+  while (not state_is_ready()
+         and deadline > ::std::chrono::system_clock::now()) {
+    log_->trace("Channel state is not ready.");
+    stub_holder_->channel_->WaitForStateChange(state, deadline);
+    state = stub_holder_->channel_->GetState(true);
+  }
+  log_->trace("Channel state is {}ready when finished waiting.",
+              state_is_ready() ? "" : "not ");
```

Comment:
log_->trace(state_is_ready() ? "Channel state is ready when finished waiting." 
                                               : "Channel state is not ready when finished waiting.");

Reply:
duplication... harder to read... (readers will waste some time to figure out the difference)